recipe-client-addon/test/browser/browser_PreferenceExperiments.js

```diff
@@ -521,41 +521,48 @@ add_task(withMockExperiments(async function (experiments) {
 // init should set the default preference value for active, default experiments
 add_task(withMockExperiments(withMockPreferences(async function testInit(experiments, mockPreferences) {
   experiments["user"] = experimentFactory({
+    name: "user",
     preferenceName: "user",
     preferenceValue: true,
     expired: false,
     preferenceBranchType: "user",
   });
   experiments["default"] = experimentFactory({
+    name: "default",
     preferenceName: "default",
     preferenceValue: true,
     expired: false,
     preferenceBranchType: "default",
   });
   experiments["expireddefault"] = experimentFactory({
+    name: "expireddefault",
     preferenceName: "expireddefault",
     preferenceValue: true,
     expired: true,
     preferenceBranchType: "default",
   });
 
-  for (const name of Object.keys(experiments)) {
-    mockPreferences.set(name, false, "default");
+  for (const experiment of Object.values(experiments)) {
+    mockPreferences.set(experiment.preferenceName, false, "default");
   }
 
   await PreferenceExperiments.init();
 
-  equal(DefaultPreferences.get("user"), false, "init ignored a user pref experiment");
-  equal(
-    DefaultPreferences.get("expireddefault"),
+  is(
+    mockPreferences.get("user", undefined, "default"),
     false,
-    "init ignored an expired default pref experiment",
+    "init ignored a user pref experiment",
   );
-  equal(
-    DefaultPreferences.get("default"),
+  is(
+    mockPreferences.get("default", undefined, "default"),
```

Comment:
It seemed strange to deal with two different objects for interacting with preferences during tests. This way a test that uses prefs only needs one object, instead of two or three (in the case of using both default and non-default prefs). It also gives us a consistent API between setting and getting prefs, which is nice.